app/src/test/java/com/eis/geoCalendar/app/JsonEventParserTest.java

```diff
@@ -0,0 +1,111 @@
+package com.eis.geoCalendar.app;
+
+import com.eis.geoCalendar.database.JsonEventParser;
+import com.eis.geoCalendar.events.Event;
+import com.google.gson.reflect.TypeToken;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Collection;
+
+import static junit.framework.TestCase.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Parameterized tests for JsonEventParser.
+ *
+ * @param <E> Event type, needed for parser and TypeToken in order to parameterize the Test on
+ *            the Event class.
+ */
+@RunWith(Parameterized.class)
+public class JsonEventParserTest<E extends Event> {
+
+    private JsonEventParser<E> eventParser;
+    private TypeToken<E> currentEventTypeToken;
+    private E currentEvent;
+    private boolean shouldAllowGeneric;
+
+    /**
+     * Method returning params for the tests. They are stored in {@link EventTestHelper}.
+     * @see EventTestHelper#nameTokenExampleGeneric() For parameter explanation.
+     * @return A collection of parameter sets suited for these tests.
+     */
+    @Parameterized.Parameters(name = "{index}: {0}")
+    public static Collection<Object[]> params() {
+        return EventTestHelper.nameTokenExampleGeneric();
+    }
+
+    /**
+     * Constructor for the Test.
+     *
+     * @param currentEventTypeToken The type of {@link Event} being tested.
+     * @param testName              Used only by {@link JsonEventParserTest#params()} to set the
+     *                              Test name.
+     * @param currentEvent          The {@link Event} being currently used to test the parser.
+     * @param shouldAllowGeneric    The {@link Event} contains such a data type that it should
+     *                              allow converting from a String even with a Raw TypeToken.
+     */
+    public JsonEventParserTest(String testName, TypeToken<E> currentEventTypeToken,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused constructor parameters such as 'testName'.](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4822079)

Reply:
I kinda need this to give the test a name, I could leave the TypeToken's name, but it gets quite verbose.